File: app/models/transfer.rb
class Transfer
  include ActiveAttr::Model
  
  attribute :user
  attribute :destination
  attribute :amount
  attribute :description
  
  validates_presence_of :user, :destination, :amount
  
  liquid_methods :user, :destination, :amount, :description
  
  def complete(till)
    @till = till
    
    if valid?
      @user = User.find_by(:id => self.user)
      @destination = Till.find_by(:id => self.destination)
      
      @adjustment = @till.adjustments.new
      @adjustment_destination = @destination.adjustments.new
      
      @adjustment.balance = @till.balance
      @adjustment.amount = self.amount.to_i * -1
      @adjustment.description = self.description
      @adjustment.title = "Transfer to #{@destination.name} #{Time.now.to_s} (#{@user.fullname})"
      @adjustment.user = @user
      
      @adjustment_destination.balance = @destination.balance
      @adjustment_destination.amount = self.amount.to_i
      @adjustment_destination.description = self.description
      @adjustment_destination.title = "Transfer from #{@till.name} #{Time.now.to_s} (#{@user.fullname})"
      @adjustment_destination.user = @user
      
      @adjustment.save && @adjustment_destination.save
    else
      false
    end
  end
end

File: app/assets/javascripts/pos/controllers/till/transfer.js
SwapzPOS.TillTransferController = Ember.ObjectController.extend({
  tills: null,
  destination: null,
  description: null,
  amount: 0,
  actions: {
    transfer: function() {
      var till = this.get('model');
      var destination = this.get('destination');
      if (till && destination) {
        var date = new Date();
        var user = till.get('user');
        var adjustment = SwapzPOS.Adjustment.create();
        var adjustment_r = SwapzPOS.Adjustment.create();
      
        adjustment.set('amount', parseInt(this.get('amount') * 100 * -1));
        adjustment.set('balance', till.get('balance'));
        adjustment.set('user', user);
        adjustment.set('description', this.get('description'));
        adjustment.set('title', "Transfer to " + destination.get('name') + " " + date + " (" + user.get('fullname') + ")");
      
        adjustment_r.set('amount', parseInt(this.get('amount') * 100));
        adjustment_r.set('balance', destination.get('balance'));
        adjustment_r.set('user', user);
        adjustment_r.set('description', this.get('description'));
        adjustment_r.set('title', "Transfer from " + till.get('name') + " " + date + " (" + user.get('fullname') + ")");
      
        till.get('adjustments').push(adjustment);
        destination.get('adjustments').push(adjustment_r);
      
        till.save(function() {
          destination.save();
        });
      }
    },
    reset: function() {
      this.set('destination', null);
      this.set('description', null);
      this.set('amount', 0);
    }
  }
});